src/types/iterator.rs

```diff
@@ -46,6 +46,28 @@ impl PyIterator {
     }
 }
 
+/// Implementation of functionality for [`PyIterator`].
+///
+/// These methods are defined for the `Bound<'py, PyIterator>` smart pointer, so to use method call
+/// syntax these methods are separated into a trait, because stable Rust does not yet support
+/// `arbitrary_self_types`.
+#[doc(alias = "PyIterator")]
+pub trait PyIteratorMethods<'py> {
+    /// Equivalent of `next(self)`.
+    fn next(&self) -> Option<PyResult<Bound<'py, PyAny>>>;
+}
```

Comment:
Actually it's probably not a good idea to have this `next()` as it'll create a name clash, will rework.

Reply:
Ok, this PR should be reviewable now.